ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Connection.scala

```diff
@@ -192,12 +192,12 @@ private[h2] class H2Connection[F[_]](
     firstGoAway.getOrElse(F.unit) >> go(chunk)
   }
 
-  def writeLoop: Stream[F, Nothing] =
-    Stream
-      .fromQueueUnterminatedChunk[F, H2Frame](outgoing, Int.MaxValue)
-      .chunks
-      .foreach(writeChunk)
-      .handleErrorWith(ex => Stream.exec(logger.debug(ex)("writeLoop terminated")))
+  def writeLoop: F[Unit] =
+    outgoing.tryTakeN(None).flatMap { (chs: List[Chunk[H2Frame]]) =>
+      val ch = Chunk.concat(chs)
+      writeChunk(ch).handleErrorWith(ex => logger.debug(ex)("writeLoop terminated")) >> writeLoop
+    }
```

Comment:
Isn't this a busy-loop?

Reply:
Busy Loop?

Reply:
What I mean is that `tryTakeN(None)` always returns immediately i.e. it will not semantically block waiting for elements in the queue. So the program will always be busy looping.

https://en.wikipedia.org/wiki/Busy_loop

Reply:
Oh, I see. I wonder, could we make `outgoing` into a `Channel`, as was done for the `body` below?